File: 3898-find-the-degree-of-each-vertex/3898-find-the-degree-of-each-vertex.js
/**
 * @param {number[][]} matrix
 * @return {number[]}
 */
var findDegrees = function (matrix) {
    const m = matrix.length;
    const n = matrix[0].length;
    const adjList = new Map();

    for (let row = 0; row < m; row++) {
        adjList.set(row, []);
    }

    for (let row = 0; row < m; row++) {
        const list = adjList.get(row);
        for (let col = 0; col < n; col++) {
            if (matrix[row][col] === 1) {
                list.push(col);
                adjList.set(row, list);
            }
        }
    }

    const result = new Array(adjList.size);
    for(let [key, val] of adjList) {
        result[key] = val.length;
    }

    return result;
};

File: 3898-find-the-degree-of-each-vertex/3898-find-the-degree-of-each-vertex.py
class Solution:
    def findDegrees(self, matrix: list[list[int]]) -> list[int]:
        m = len(matrix)
        n = len(matrix[0])
        adjList = {}
        
        for row in range(m):
            for col in range(n):
                if matrix[row][col] == 1:
                    list = adjList.get(row, [])
                    list.append(col)
                    adjList[row] = list

        result = [0] * m
        for key, val in adjList.items():
            result[key] = len(val)

        return result